lnwallet/chainfee/estimator.go

```diff
@@ -593,8 +600,12 @@ func (w *WebAPIEstimator) EstimateFeePerKW(numBlocks uint32) (SatPerKWeight, err
 	}
 
 	feePerKb, err := w.getCachedFee(numBlocks)
+
+	// If the estimator returns an error, a zero value fee rate will be
+	// returned. We will log the error and return the fall back fee rate
+	// instead.
 	if err != nil {
-		return 0, err
+		log.Errorf("unable to query estimator: %v", err)
```

Comment:
I'm not sure returning the default value (fee floor!) is a good idea here. The problem is that the code that generates the https://nodes.lightning.computer/fees/v1/btc-fee-estimates.json file is never going to include a fee estimate for a 1 block confirmation target (it's using https://github.com/lightningnetwork/lnd/blob/master/lnrpc/walletrpc/walletkit_server.go#L636).
And since a 1 block target is the "most urgent fee" setting, returning the _minimum_ fee is probably the worst thing we can do. Should we try to return the fee value for the _next_ higher confirmation target? In this case for 2 blocks? And only if there is no value at all return the fee floor?

Reply:
IIRC bitcoind (on the RPC interface) never allows a 1 conf query, and instead only allows 2 blocks as a minimum

Reply:
IMO we should just add a special case for conf target of 1 here. And simply try to return the value for conf target of 2.

Reply:
> IIRC bitcoind (on the RPC interface) never allows a 1 conf query, and instead only allows 2 blocks as a minimum

Tested with `bitcoin-cli estimatesmartfee 1` and did pass. However I think it falled back to conf target of 2.

Reply:
Hmm, maybe the issue is that we don't use the `estimatesmartfee` call yet, but an older one?

Reply:
> IMO we should just add a special case for conf target of 1 here.

Will take this approach. So split it into three cases, when a conf target cannot be found,
a) try search a lower conf, which will give a more expensive fee rate and we will log a warning there
b) if the above fails, try search a larger conf, which will give a cheaper fee rate and we will log an error, hopefully users will catch it
c) if all the above fail, return the floor fee rate and log an error, this is a very unlikely case unless the web API has nothing in it